### PR TITLE
LoRaCloud GLS GNSS support

### LINKED ISSUE #4032: GNSS query support for LoRaCloud GLS
#### Summary

References https://www.loracloud.com/documentation/geolocation?url=gnss.html

#### Why do we need this?

In order to support GNSS queries as part of our integration.

#### What is already there? What do you see now?

Only v3, single-frame, queries using `TOARSSI`.

#### What is missing? What do you want to see?

1. A new query type, `GNSS`, that should do a v3 GNSS request.
2. `api` package support for this type of request.

#### Environment

`v3.12`

#### How do you propose to implement this?

Add the new query type, implement the GNSS request query (so structs + API function), and use it on uplink handling.

#### How do you propose to test this?

This is tricky. We'll need to lookup some example queries and use them.

#### Can you do this yourself and submit a Pull Request?

Yes.

### PR DESCRIPTION
**Pull request by adriansmares**
#### Summary

Closes #4032 

#### Changes

- Add GNSS API support and updated tests
- Add GNSS support to the LoRaCloud GLS package
- Create a unified interface for the location query responses, as the APIs diverge

#### Testing

Local testing with static GNSS frames.

#### Notes for Reviewers

As before, it targets a feature branch in order to isolate only relevant (and unreviewed) commits.

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [ ] Documentation: Relevant documentation is added or updated.
- [x] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

**Review by asmulko — APPROVED** [review]

**Review by bafonins — APPROVED** [review]
`pkg/webui` changes look fine

**Comment by adriansmares** [comment]
Rebased in order to test the validation changes from the target branch. Working as expected.

File: pkg/applicationserver/io/packages/loragls/v3/data.go [hunk 1]
```diff
@@ -64,6 +68,8 @@ func (t *QueryType) FromValue(v *types.Value) error {
 const (
 	// QUERY_TOARSSI uses the TOA and RSSI information from the gateway metadata to compute the location of the end device.
 	QUERY_TOARSSI QueryType = 1 << 0
+	// QUERY_GNSS uses the GNSS scan operations payload of the LR1110 transceiver.
+	QUERY_GNSS = 1 << 1
```
Comment (johanstokking) [inline pkg/applicationserver/io/packages/loragls/v3/data.go]:
```suggestion
	// QUERY_TOARSSI uses the TOA and RSSI information from the gateway metadata to compute the location of the end device.
	QUERY_TOARSSI QueryType = 1 << iota
	// QUERY_GNSS uses the GNSS scan operations payload of the LR1110 transceiver.
	QUERY_GNSS
```

**Review by johanstokking — APPROVED** [review]

File: pkg/webui/console/containers/lora-cloud-gls-form/index.js [hunk 2]
```diff
@@ -62,9 +62,13 @@ const m = defineMessages({
   multiFrameTimeWindowDescription: 'The maximum age of considered historical messages in minutes.',
```
Comment (kschiffer) [inline pkg/webui/console/containers/lora-cloud-gls-form/index.js]:
```suggestion
  multiFrameTimeWindowDescription: 'The maximum age of considered historical messages in minutes',
```

**Review by kschiffer — APPROVED** [review]
Codeowned LGTM.

**Pull request merged**